fs/inode/fs_files.c

```diff
@@ -82,26 +82,26 @@ static FAR struct file *files_fget_by_index(FAR struct filelist *list,
        * released, At this point we should return a null pointer
        */
 
-      if (filep->f_refs == 0)
+      if (atomic_read(&filep->f_refs) == 0)
         {
           filep = NULL;
         }
       else
         {
-          filep->f_refs++;
+          atomic_fetch_add(&filep->f_refs, 1);
         }
     }
   else if (new == NULL)
     {
       filep = NULL;
     }
-  else if (filep->f_refs)
+  else if (atomic_read(&filep->f_refs))
     {
-      filep->f_refs++;
+      atomic_fetch_add(&filep->f_refs, 1);
```

Comment:
race condition ... if ISR after read f_refs

Reply:
These few lines of code are already being executed with irq disabled.

Reply:
1. [CPU1]filep->f_refs == 1
2. [CPU1]atomic_read > 0
3. [CPU2]atomic_fetch_sub - 1 (fs_putfilep)
5. [CPU1]atomic_fetch_add + 1
4. [CPU2]file_close

Reply:
The issue you mentioned was introduced by me. Here's the fix:
https://github.com/apache/nuttx/pull/15163